client/core/types.go

```diff
@@ -105,35 +105,75 @@ type RegisterForm struct {
+		Status:  userMatch.Status,
+		Rate:    userMatch.Rate,
+		Qty:     userMatch.Quantity,
+		Side:    userMatch.Side,
+		FeeRate: userMatch.FeeRateSwap,
+		Refund:  []byte(proof.RefundCoin),
+		Stamp:   metaMatch.MetaData.Stamp,
+	}
+
+	if userMatch.Side == order.Maker {
+		match.Swap = []byte(proof.MakerSwap)
+		match.CounterSwap = []byte(proof.TakerSwap)
+		match.Redeem = []byte(proof.MakerRedeem)
+		match.CounterRedeem = []byte(proof.TakerRedeem)
+	} else {
+		match.Swap = []byte(proof.TakerSwap)
+		match.CounterSwap = []byte(proof.MakerSwap)
+		match.Redeem = []byte(proof.TakerRedeem)
+		match.CounterRedeem = []byte(proof.MakerRedeem)
+	}
+
+	return match
 }
 
 // Order is core's general type for an order. An order may be a market, limit,
 // or cancel order. Some fields are only relevant to particular order types.
 type Order struct {
-	Host        string            `json:"host"`
-	MarketID    string            `json:"market"`
-	Type        order.OrderType   `json:"type"`
-	ID          string            `json:"id"`
-	Stamp       uint64            `json:"stamp"`
-	Sig         dex.Bytes         `json:"sig"`
-	Status      order.OrderStatus `json:"status"`
-	Epoch       uint64            `json:"epoch"`
-	Qty         uint64            `json:"qty"`
-	Sell        bool              `json:"sell"`
-	Filled      uint64            `json:"filled"`
-	Matches     []*Match          `json:"matches"`
-	Cancelling  bool              `json:"cancelling"`
-	Canceled    bool              `json:"canceled"`
-	FeesPaid    *FeeBreakdown     `json:"feesPaid"`
-	Rate        uint64            `json:"rate"`               // limit only
-	TimeInForce order.TimeInForce `json:"tif"`                // limit only
-	TargetID    string            `json:"targetID,omitempty"` // cancel only
+	Host         string            `json:"host"`
+	BaseID       uint32            `json:"baseID"`
+	BaseSymbol   string            `json:"baseSymbol"`
+	QuoteID      uint32            `json:"quoteID"`
+	QuoteSymbol  string            `json:"quoteSymbol"`
+	MarketID     string            `json:"market"`
+	Type         order.OrderType   `json:"type"`
+	ID           dex.Bytes         `json:"id"`
+	Stamp        uint64            `json:"stamp"`
+	Sig          dex.Bytes         `json:"sig"`
+	Status       order.OrderStatus `json:"status"`
+	Epoch        uint64            `json:"epoch"`
+	Qty          uint64            `json:"qty"`
+	Sell         bool              `json:"sell"`
+	Filled       uint64            `json:"filled"`
+	Matches      []*Match          `json:"matches"`
+	Cancelling   bool              `json:"cancelling"`
+	Canceled     bool              `json:"canceled"`
+	FeesPaid     *FeeBreakdown     `json:"feesPaid"`
+	FundingCoins []dex.Bytes       `json:"fundingCoins"`
+	Rate         uint64            `json:"rate"`               // limit only
+	TimeInForce  order.TimeInForce `json:"tif"`                // limit only
+	TargetID     dex.Bytes         `json:"targetID,omitempty"` // cancel only
```

Comment:
New fields are `BaseID`, `BaseSymbol`, `QuoteID`, `QuoteSymbol`, and `FundingCoins`.